truffle/src/main/java/org/jruby/truffle/nodes/rubinius/IOPrimitiveNodes.java

```diff
@@ -43,4 +89,100 @@ public boolean fnmatch(RubyString pattern, RubyString path, int flags) {
 
     }
 
+    @RubiniusPrimitive(name = "io_ensure_open")
+    public static abstract class IOEnsureOpenPrimitiveNode extends RubiniusPrimitiveNode {
+
+        public IOEnsureOpenPrimitiveNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public IOEnsureOpenPrimitiveNode(IOEnsureOpenPrimitiveNode prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public RubyNilClass ensureOpen(RubyBasicObject file) {
+            // TODO CS 18-Apr-15
+            return nil();
+        }
+
+    }
+
+    @RubiniusPrimitive(name = "io_write")
+    public static abstract class IOWritePrimitiveNode extends RubiniusPrimitiveNode {
+
+        public IOWritePrimitiveNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public IOWritePrimitiveNode(IOWritePrimitiveNode prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public int write(VirtualFrame frame, RubyBasicObject file, RubyString string) {
+            final int fd = (int) rubyWithSelf(frame, file, "@descriptor");
+
+            // We have to copy here as write starts at byte[0], and the ByteList may not
```

Comment:
We really need a pointer to the middle of the array.
Maybe using the write version with a ByteBuffer can do that?